crates/matrix-sdk-ui/src/timeline/event_item/content/mod.rs

```diff
@@ -333,19 +333,29 @@ impl TimelineItemContent {
     /// If `self` is of the [`Aggregated`][Self::Aggregated] variant, return the
     /// inner [`Message`].
     pub fn as_message(&self) -> Option<&Message> {
-        let aggregated = as_variant!(self, Self::Aggregated)
-            .map(|f| as_variant!(&f.kind, AggregatedTimelineItemContentKind::Message));
-
-        aggregated.unwrap_or(None)
+        as_variant!(
+            self,
+            Self::Aggregated(
+                AggregatedTimelineItemContent {
+                    kind: AggregatedTimelineItemContentKind::Message(msg),
+                    ..
+                }
+            ) => msg
+        )
     }
 
     /// If `self` is of the [`Aggregated`][Self::Aggregated] variant, return the
     /// inner [`PollState`].
     pub fn as_poll(&self) -> Option<&PollState> {
-        let aggregated = as_variant!(self, Self::Aggregated)
-            .map(|f| as_variant!(&f.kind, AggregatedTimelineItemContentKind::Poll));
-
-        aggregated.unwrap_or(None)
+        as_variant!(
+            self,
+            Self::Aggregated(
+                AggregatedTimelineItemContent {
+                    kind: AggregatedTimelineItemContentKind::Poll(poll),
+                    ..
+                }
+            ) => poll
+        )
```

Comment:
Potential alternative:

```rust
let aggregated = as_variant!(self, Self::Aggregated)?;
as_variant!(aggregated.kind, AggregatedTimelineItemContentKind::Poll)
```

Reply:
This variant improves the readability indeed.

Reply:
Yeah, that would've been equivalent to the simplified previous implementation. I don't have strong opinions here; OTOH in like the dense pattern matching that shows precisely what we extract where; OTOH, rustfmt didn't format it automatically, and a single-line construct was super long to read.

Reply:
Well I guess we didn't need the new pattern matching rule then :D Will update to the shorter form.